### ISSUE TITLE
`W0612 Unused variable` false positives on class within module

### ISSUE THREAD

**Issue by CMoncur** [PyCQA/pylint#1880]
### Steps to reproduce
1. Create `example.py` script with the following code:
```python
""" Example """

class Hello:
  """ Print stuff """

  def __init__(self, hi_text, bye_text):
    self.hi = hi_text
    self.bye = bye_text

  def printHi(self):
    """ Prints 'hi' """
    print(self.hi)

  def printBye(self):
    """ Prints 'bye' """
    print(self.bye)

```
2. Run `pylint example.py` in terminal
3. Observe results

**NOTE:**
I've read the similar issue tickets, and my results seem to be inconsistent with those, as instead of `Unused variable '__class__'` as the text, my results list the actual class name in the error (`Unused variable 'Hello'`).

In addition, I've attempted to roll back my astroid version to 1.5.3, as suggested in https://github.com/PyCQA/pylint/issues/1784 to no avail.

Lastly, in my pylintrc, I have the following two rules disabled: 
- `C0103`
- `C0326`

### Current behavior
- This is the current output from `pylint example.py`:
```
************* Module proto_scrape.example
W:  3, 0: Unused variable 'Hello' (unused-variable)

------------------------------------------------------------------
Your code has been rated at 8.75/10 (previous run: 8.75/10, +0.00)
```

### Expected behavior
- No error/warning should be present. The example given should produce no linter issues.

### pylint --version output
```
pylint 1.8.2,
astroid 1.5.3
Python 3.6.4 (default, Jan  6 2018, 11:51:59)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

**Comment by brycepg** [comment]
I'm not able to reproduce this with:

```
pylint 1.8.2, 
astroid 1.5.3
Python 3.6.4 (default, Feb 14 2018, 22:42:47) 
[GCC 7.2.1 20170915 (Red Hat 7.2.1-2)]
```

**Comment by PCManticore** [comment]
I can't reproduce this one as well. Also pylint 1.8 should go with astroid 1.6 (we pin it in our requirements here https://github.com/PyCQA/pylint/blob/1.8/pylint/__pkginfo__.py#L34, not sure how you two got 1.5.3 instead)

**Comment by CMoncur** [comment]
@PCManticore I used 1.5.3 per your recommended fix in https://github.com/PyCQA/pylint/issues/1784.

More importantly, I've re-updated astroid, my `pylint --version` now reads:
```
pylint 1.8.2,
astroid 1.6.1
Python 3.6.4 (default, Jan  6 2018, 11:51:59)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

And when running `pylint example.py` I'm still getting:
```
************* Module proto_scrape.example
W:  3, 0: Unused variable 'Hello' (unused-variable)

------------------------------------------------------------------
Your code has been rated at 8.75/10 (previous run: 8.75/10, +0.00)
```

**Comment by brycepg** [comment]
@PCManticore I did `pip install astroid==1.5.3 pylint==1.8.2`

@CMoncur Could you can set up a github repository which reproduces this issue with the `proto_scrape` (or similar dummy) package and your pylintrc?

**Comment by CMoncur** [comment]
@PCManticore @brycepg 
I compared my pylintrc with a newly generated one, and the culprit is the `allow-global-unused-variables` flag, which I had set to `no`. I apologize, this was entirely my mistake. This ticket can be closed -- working as intended.
